src/routes/safe/store/actions/transactions/utils/transactionHelpers.ts

```diff
@@ -0,0 +1,375 @@
+    .set('type', calculateTransactionType(txToStore))
+}
+
+export const mockTransaction = (tx, safeAddress: string, state): Promise<any> => {
+  const submissionDate = new Date().toISOString()
+
+  const transactionStructure: TxServiceModel = {
+    blockNumber: null,
+    confirmationsRequired: null,
+    dataDecoded: decodeMethods(tx.data),
+    ethGasPrice: null,
+    executionDate: null,
+    executor: null,
+    fee: null,
+    gasUsed: null,
+    isExecuted: false,
+    isSuccessful: null,
+    modified: submissionDate,
+    origin: null,
+    safe: safeAddress,
+    safeTxHash: null,
+    signatures: null,
+    submissionDate,
+    transactionHash: null,
+    confirmations: [],
+    ...tx,
+  }
+
+  const knownTokens = state[TOKEN_REDUCER_ID]
+  const safe = state[SAFE_REDUCER_ID].getIn([SAFE_REDUCER_ID, safeAddress])
+  const cancellationTxs = state[CANCELLATION_TRANSACTIONS_REDUCER_ID].get(safeAddress) || Map()
+  const outgoingTxs = state[TRANSACTIONS_REDUCER_ID].get(safeAddress) || List()
+
+  return buildTx({
+    cancellationTxs,
+    currentUser: null,
+    knownTokens,
+    outgoingTxs,
+    safe,
+    tx: transactionStructure,
+    txCode: EMPTY_DATA,
+  })
+}
+
+export const updateStoredTransactionsStatus = (dispatch, walletRecord): void => {
+  const state = store.getState()
+  const safe = safeSelector(state)
+
+  if (safe) {
+    const safeAddress = safe.address
+    const transactions = safeTransactionsSelector(state)
+    dispatch(
+      addOrUpdateTransactions({
+        safeAddress,
+        transactions: transactions.withMutations((list) =>
+          list.map((tx) => tx.set('status', calculateTransactionStatus(tx, safe, walletRecord.account))),
+        ),
+      }),
+    )
+  }
+}
+
+export function generateSafeTxHash(safeAddress: string, txArgs: TxArgs): string {
+  const typedData = {
+    types: {
+      EIP712Domain: [{ type: 'address', name: 'verifyingContract' }],
+      SafeTx: [
+        { type: 'address', name: 'to' },
+        { type: 'uint256', name: 'value' },
+        { type: 'bytes', name: 'data' },
+        { type: 'uint8', name: 'operation' },
+        { type: 'uint256', name: 'safeTxGas' },
+        { type: 'uint256', name: 'baseGas' },
+        { type: 'uint256', name: 'gasPrice' },
+        { type: 'address', name: 'gasToken' },
+        { type: 'address', name: 'refundReceiver' },
+        { type: 'uint256', name: 'nonce' },
+      ],
+    },
+    domain: {
+      verifyingContract: safeAddress,
+    },
+    primaryType: 'SafeTx',
+    message: {
+      to: txArgs.to,
+      value: txArgs.valueInWei,
+      data: txArgs.data,
+      operation: txArgs.operation,
+      safeTxGas: txArgs.safeTxGas,
+      baseGas: txArgs.baseGas,
+      gasPrice: txArgs.gasPrice,
+      gasToken: txArgs.gasToken,
+      refundReceiver: txArgs.refundReceiver,
+      nonce: txArgs.nonce,
+    },
+  }
+
+  // @ts-ignore
+  return `0x${TypedDataUtils.sign(typedData).toString('hex')}`
```

Comment:
`sign` is not accepting the structure of `typedData`. Thus, I added `@ts-ignore`, but if there's a correct way to make it pass without ignoring `ts`, please tell me.

Reply:
I've fixed it, had to look into the source code of `eth-sig-util`. They have some magic with generics & type arguments: https://github.com/MetaMask/eth-sig-util/blob/master/index.ts#L40

In general we should never use `@ts-ignore` (also `any` type) because typescript strict mode/eslint config won't allow that

Reply:
thank you!!